core/src/test/java/org/springframework/security/access/expression/method/DefaultMethodSecurityExpressionHandlerTests.java

```diff
@@ -68,4 +72,22 @@ public void createEvaluationContextCustomTrustResolver() {
 
 		verify(trustResolver).isAnonymous(authentication);
 	}
+
+	@Test
+	@SuppressWarnings("unchecked")
+	public void testFilteringStream() {
```

Comment:
Do you have time to add just one more test? Because @nightswimmings brought up the fact that we need to preserve the closeability of the original stream, it would be good to prove that that works in a unit test. Something like:

```suggestion
	@Test
	public void filterWhenClosedThenUpstreamGetsClosed() {
		Stream<String> upstream = mock(Stream.class);

		// use expression handler to get filtered stream

		filtered.close();
		verify(upstream).close();
	}
```